pkg/cmd/pr/pr.go

```diff
@@ -165,6 +167,8 @@ func (o *Options) Run() error {
 
 			// lets clear the branch name so we create a new one each time in a loop
 			o.BranchName = ""
+			// lets reset the base branch name each time in a loop to avoid side effects when reusing pull requests
+			o.BaseBranchName = BaseBranchName
```

Comment:
it would be good to write a test case, since u found a bug.

Reply:
@ankitm123 It requires an actual git environment with multiple urls to reproduce this error and I am not very experienced in Go testing yet. This problem broke propagation in our pipelines, so if you can ok to test my PR it will be much appreciated.

Reply:
ok, I can write the test after kubecon next week. It would be good if you can give me a simple reproduction, and I can try to reproduce the issue in unit tests.

For now, I will approve it.